lib/membrane_overlay_filter.ex

```diff
@@ -19,48 +21,28 @@ defmodule Membrane.OverlayFilter do
   alias Vix.Vips.Image, as: Vimage
   alias Vix.Vips.Operation
 
-  def_options overlay: [
-                spec: Path.t() | Vix.Vips.Image.t(),
+  def_options initial_overlay: [
+                spec: OverlayDescription.t(),
                 description: """
-                Path to the overlay image or a `Vix` image.
-
-                You can get a `Vix` image for example by calling `Image.open/2`,
-                `Image.Text.text/2` or `Vix.Vips.Image.new_from_buffer/2`.
-                """
-              ],
-              x: [
-                spec: integer() | :center | :left | :right,
-                default: :center,
-                description: """
-                Distance of the overlay image from the left (or right if negative)
-                border of the frame. Can be also set to center, left or right.
-                """
-              ],
-              y: [
-                spec: integer() | :middle | :top | :bottom,
-                default: :middle,
-                description: """
-                Distance of the overlay image from the top (or bottom if negative)
-                border of the frame. Can be also set to middle, top or bottom.
-                """
-              ],
-              blend_mode: [
-                spec: Image.BlendMode.t(),
-                default: :over,
-                description: """
-                The manner in which the overlay is composed on the frame.
+                Description of the overlay that will be initially used.
                 """
               ]
 
   @impl true
   def handle_init(_ctx, options) do
-    opts_y = options |> Map.take([:x, :y, :blend_mode]) |> Enum.to_list()
-    uv_x = if is_integer(options.x), do: div(options.x, 2), else: options.x
-    uv_y = if is_integer(options.y), do: div(options.y, 2), else: options.y
-    opts_uv = [x: uv_x, y: uv_y, blend_mode: options.blend_mode]
+    initial_overlay = options.initial_overlay
+    opts_y = initial_overlay |> Map.take([:x, :y, :blend_mode]) |> Enum.to_list()
+
+    uv_x =
+      if is_integer(initial_overlay.x), do: div(initial_overlay.x, 2), else: initial_overlay.x
+
+    uv_y =
+      if is_integer(initial_overlay.y), do: div(initial_overlay.y, 2), else: initial_overlay.y
+
+    opts_uv = [x: uv_x, y: uv_y, blend_mode: initial_overlay.blend_mode]
```

Comment:
I would just take out this logic into a separate function, because it's repeated here and in `handle_parent_notification`